src/Common/tests/System/Collections/IEnumerable.Generic.Tests.cs

```diff
@@ -37,7 +37,11 @@ public abstract partial class IEnumerable_Generic_Tests<T> : TestBase<T>
         /// To be implemented in the concrete collections test classes. Returns a set of ModifyEnumerable delegates
         /// that modify the enumerable passed to them.
         /// </summary>
-        protected abstract IEnumerable<ModifyEnumerable> ModifyEnumerables { get; }
+        protected abstract IEnumerable<ModifyEnumerable> GetModifyEnumerables(ModifyOperation operations);
+
+        protected virtual ModifyOperation ModifyEnumeratorThrows => ModifyOperation.Add | ModifyOperation.Insert | ModifyOperation.Remove | ModifyOperation.Clear;
+
+        protected virtual ModifyOperation ModifyEnumeratorAllowed => ModifyOperation.None;
```

Comment:
Is `ModifyEnumeratorAllowed` necessary - couldn't it just check what is not in `ModifyEnumeratorThrows` ?

Reply:
It could be done without. If you assume that all operations, present and future, must be either allowed or disallowed. I thought perhaps someone might want to add an operation such as Set which may not apply to all containers from IEnumerable upwards and that making this explicit now may save some pain later.